src/agentscope/strategy/mixture_of_agent.py

```diff
@@ -0,0 +1,171 @@
+        self.reference_models: List[ModelWrapperBase] = []
+        for ref_model in reference_models:
+            if isinstance(ref_model, str):
+                self.reference_models.append(
+                    load_model_by_config_name(ref_model),
+                )
+            elif isinstance(ref_model, ModelWrapperBase):
+                self.reference_models.append(ref_model)
+            else:
+                raise ValueError(
+                    "reference_models must be a list of strings "
+                    "or ModelWrapperBase instances",
+                )
+        self.references: List[str] = [
+            "" for _ in range(len(self.reference_models))
+        ]
+        self.rounds = rounds
+        self.aggregator_prompt = aggregator_prompt
+        self.show_internal = show_internal
+
+    def _get_res_with_aggregate_model(
+        self,
+        aggre_model: ModelWrapperBase,
+    ) -> str:
+        messages = []
+        messages.append(
+            Msg(role="system", content=self.aggregator_prompt, name="system"),
+        )
+        for i, ref in enumerate(self.references, start=0):
+            messages.append(
+                Msg(
+                    role="user",
+                    content=ref,
+                    name=f"Model_{i}",
+                ),
+            )
+        aggre_format_msg = aggre_model.format(messages)
+        aggre_res = aggre_model(aggre_format_msg)
+        return aggre_res.text
+
+    def __call__(
+        self,
+        *args: Union[Msg, Sequence[Msg]],
+    ) -> str:
+        """
+        Get model response from messages.
+        Is equivalent to calling a model with:
+            ```
+            format_msg = model.format(messages)
+            return model(format_msg)
+            ```
+
+        Args:
+            *args (`Union[Msg, Sequence[Msg]]`):
+                The messages to be sent to the model.
+        """
+
+        def _process_reference(
+            i: int,
+            ref_model: ModelWrapperBase,
+            *args: Union[Msg, Sequence[Msg]],
+        ) -> Tuple[int, str]:
+            format_msg = ref_model.format(*args)
+            ref_model_res = ref_model(format_msg)
+            return i, ref_model_res.text
+
+        def _process_new_refs(
+            i: int,
+            ref_model: ModelWrapperBase,
+        ) -> Tuple[int, str]:
+            return i, self._get_res_with_aggregate_model(ref_model)
+
+        # get all the references
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            futures = [
+                executor.submit(_process_reference, i, ref_model, *args)
+                for i, ref_model in enumerate(self.reference_models, start=0)
+            ]
+            for future in concurrent.futures.as_completed(futures):
+                i, result = future.result()
+                self.references[i] = result
+                if self.show_internal:
+                    print(f"Round {0}, Model_{i}: {result}")
```

Comment:
Consider to use logger.info rather than print here to record the internal information.